tutorials/micro/micro_autotune.py

```diff
@@ -0,0 +1,271 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+"""
+.. _tutorial-micro-autotune:
+
+Autotuning with micro TVM
+=========================
+**Author**: `Andrew Reusch <https://github.com/areusch>`_, `Mehrdad Hessar <https://github.com/mehrdadh>`
+
+This tutorial explains how to autotune a model using the C runtime.
+"""
+
+import argparse
+
+# A mapping of a microTVM device to its target and board.
+PLATFORMS = {
+    "host": ("host", None),
+    "qemu_x86": ("host", "qemu_x86"),
+    "nrf5340dk": ("nrf5340dk", "nrf5340dk_nrf5340_cpuapp"),
+    "stm32f746xx_disco": ("stm32f746xx", "stm32f746g_disco"),
+    "stm32f746xx_nucleo": ("stm32f746xx", "nucleo_f746zg"),
+    "stm32l4r5zi_nucleo": ("stm32l4r5zi", "nucleo_l4r5zi"),
+}
+
+
+def main(args):
+    ####################
+    # Defining the model
+    ####################
+    #
+    # To begin with, define a model in Relay to be executed on-device. Then create an IRModule from relay model and
+    # fill parameters with random numbers.
+    #
+
+    import tvm
+
+    data_shape = (1, 3, 10, 10)
+    weight_shape = (6, 3, 5, 5)
+
+    data = tvm.relay.var("data", tvm.relay.TensorType(data_shape, "float32"))
+    weight = tvm.relay.var("weight", tvm.relay.TensorType(weight_shape, "float32"))
+
+    y = tvm.relay.nn.conv2d(
+        data,
+        weight,
+        padding=(2, 2),
+        kernel_size=(5, 5),
+        kernel_layout="OIHW",
+        out_dtype="float32",
+    )
+    f = tvm.relay.Function([data, weight], y)
+
+    relay_mod = tvm.IRModule.from_expr(f)
+    relay_mod = tvm.relay.transform.InferType()(relay_mod)
+
+    import numpy as np
```

Comment:
Is there a reason to import numpy in the function vs at the top of the module? If so, could you comment it?

Reply:
no reason. Just did it follow our tutorial style where we import stuff immediately after we explain a section too keep things in isolated block. This was my impression, I can change it if you think.